app/controllers/tracks_controller.rb

```diff
@@ -19,7 +19,10 @@ def index
   def show
     @user_track = UserTrack.for(current_user, @track)
 
-    if @user_track
+    if @user_track.external?
+      @showcase_exercises = @user_track.exercises.order("RAND()").limit(3).to_a
+      render "tracks/show/unjoined"
+    else
```

Comment:
```suggestion
      return render "tracks/show/unjoined"
    end
```

(and remove the end below)